packages/client/src/runtime/core/request/PrismaPromise.ts

```diff
@@ -3,23 +3,16 @@
  * original `Promise` are optional so that it can be backwards-compatible.
  * @see [[createPrismaPromise]]
  */
-
 export interface PrismaPromise<A> extends Promise<A> {
   /**
    * Extension of the original `.then` function
    * @param onfulfilled same as regular promises
    * @param onrejected same as regular promises
-   * @param transactionId for interactive tx ids
+   * @param txId string for iTX & Promise for regular tx
    */
   then<R1 = A, R2 = never>(
     onfulfilled?: (value: A) => R1 | PromiseLike<R1>,
     onrejected?: (error: unknown) => R2 | PromiseLike<R2>,
-    transactionId?: number,
+    txId?: string | PromiseLike<number>,
   ): Promise<R1 | R2>
-
-  /**
-   * Called when executing a batch of regular tx
-   * @param id for regular tx ids
-   */
-  requestTransaction?(id: number): PromiseLike<unknown>
```

Comment:
Technically, this renaming from `transactionId` to `txId` and type change could be a breaking change for users using `$transaction` which is GA?

Example
https://github.com/juanluispaz/ts-sql-query/blob/0d73da992a8f8ce396044f75fe51095aea2a2ed5/src/queryRunners/PrismaQueryRunner.ts#L286-L337

Also how can it be a string now?

Reply:
Example
https://github.com/juanluispaz/ts-sql-query/blob/0d73da992a8f8ce396044f75fe51095aea2a2ed5/src/queryRunners/PrismaQueryRunner.ts#L286-L337

Reply:
Also the removal of `requestTransaction` (to consider as breaking change)

Reply:
about the type string I moved conversation to https://github.com/prisma/prisma/pull/11695/files#r811005220

Reply:
Fixed